.changeset/glorious-turtles-try.md

```diff
@@ -0,0 +1,22 @@
+---
+"@fluidframework/test-end-to-end-tests": major
+"@fluidframework/container-runtime": major
+"@fluid-internal/test-service-load": major
```

Comment:
```suggestion
"@fluidframework/test-end-to-end-tests": minor
"@fluidframework/container-runtime": minor
"@fluid-internal/test-service-load": minor
```

These should be minor if it's going into main.

Reply:
@tylerbutler Is there a reason it got generated as `major` instead of `minor`?

Reply:
@kian-thompson I chose that option when creating the change set, because I thought this was going into next. If I understand correctly, this is a new feature/behavior that is behind a feature flag, so it's safe to make the change in main? And we'd announce the feature in the version it is available in, even if it's off by default, then when we turn it on then we'd announce that change as well (presumably a major version).

Reply:
This change adds new file format (main thing that happens here). But as any format change, it has to ship dark, so there is no change in behavior at the moment.
We will keep iterating on it, so it may be many versions before we feel comfortable to give green light (via allowing people to opt-in, or make it ON by default).
Apps can effectively use new behavior only when app developer confirms 99.99% of app sessions are on a build that contain these changes (and only after we remove experimental marker on it). That time will be different for every app.

Basically, format change lifecycle is orthogonal to versions of FF.

Reply:
> Basically, format change lifecycle is orthogonal to versions of FF.

Do we need a separate place outside of release announcements/changelog to talk about these types of features since they don't follow a release-oritented timeline? Have we expected people to watch BREAKING.md to get this information in the past?

Reply:
Maybe?

Current flow - we announce the option when the code is in. This is optional, we can skip it as we expect no action on behalf of users, and option should be marked as experimental (and in-place documentation should talk about danger of enabling it).

Then we announce when we flip default to On, and clearly communicate requirement for how to safely consume such change. It is breaking change at that moment (as consuming & shipping it to PROD requires app developer to analyze if it's safe to do so, this decision is app specific).

So, it feels like we do not need change in the flow for first part, and regular major process is sufficient for later? The only difference for later is that app developers can't go with "if it builds and passes tests - it's good" approach